pkg/apiserver/schedule/schedule.go

```diff
@@ -835,3 +838,130 @@ func (s *Service) batchDeleteSchedule(c *gin.Context) {
 		c.JSON(http.StatusOK, StatusResponse{Status: "success"})
 	}
 }
+
+// @Summary Pause a schedule experiment.
+// @Description Pause a schedule experiment.
+// @Tags schedules
+// @Produce json
+// @Param uid path string true "uid"
+// @Success 200 {object} StatusResponse
+// @Failure 400 {object} utils.APIError
+// @Failure 404 {object} utils.APIError
+// @Failure 500 {object} utils.APIError
+// @Router /schedules/pause/{uid} [put]
+func (s *Service) pauseSchedule(c *gin.Context) {
+	var schedule *core.Schedule
+
+	kubeCli, err := clientpool.ExtractTokenAndGetClient(c.Request.Header)
+	if err != nil {
+		_ = c.Error(utils.ErrInvalidRequest.WrapWithNoMessage(err))
+		return
+	}
+
+	uid := c.Param("uid")
+	if schedule, err = s.schedule.FindByUID(context.Background(), uid); err != nil {
+		if gorm.IsRecordNotFoundError(err) {
+			c.Status(http.StatusInternalServerError)
+			_ = c.Error(utils.ErrInvalidRequest.New("the schedule is not found"))
+		} else {
+			c.Status(http.StatusInternalServerError)
+			_ = c.Error(utils.ErrInternalServer.NewWithNoMessage())
+		}
+		return
+	}
+
+	exp := &Base{
+		Name:      schedule.Name,
+		Namespace: schedule.Namespace,
+	}
+
+	annotations := map[string]string{
+		v1alpha1.PauseAnnotationKey: "true",
+	}
+	if err := s.patchSchedule(exp, annotations, kubeCli); err != nil {
+		if apierrors.IsNotFound(err) {
+			c.Status(http.StatusNotFound)
+			_ = c.Error(utils.ErrNotFound.WrapWithNoMessage(err))
+			return
+		}
+		c.Status(http.StatusInternalServerError)
+		_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
+		return
+	}
+
+	c.JSON(http.StatusOK, StatusResponse{Status: "success"})
+}
+
+// @Summary Start a schedule experiment.
+// @Description Start a schedule experiment.
+// @Tags schedules
+// @Produce json
+// @Param uid path string true "uid"
+// @Success 200 {object} StatusResponse
+// @Failure 400 {object} utils.APIError
+// @Failure 404 {object} utils.APIError
+// @Failure 500 {object} utils.APIError
+// @Router /schedules/start/{uid} [put]
+func (s *Service) startSchedule(c *gin.Context) {
+	var schedule *core.Schedule
+
+	kubeCli, err := clientpool.ExtractTokenAndGetClient(c.Request.Header)
+	if err != nil {
+		_ = c.Error(utils.ErrInvalidRequest.WrapWithNoMessage(err))
+		return
+	}
+
+	uid := c.Param("uid")
+	if schedule, err = s.schedule.FindByUID(context.Background(), uid); err != nil {
+		if gorm.IsRecordNotFoundError(err) {
+			c.Status(http.StatusInternalServerError)
+			_ = c.Error(utils.ErrInvalidRequest.New("the experiment is not found"))
+		} else {
+			c.Status(http.StatusInternalServerError)
+			_ = c.Error(utils.ErrInternalServer.NewWithNoMessage())
+		}
+		return
+	}
+
+	exp := &Base{
+		Name:      schedule.Name,
+		Namespace: schedule.Namespace,
+	}
+
+	annotations := map[string]string{
+		v1alpha1.PauseAnnotationKey: "false",
```

Comment:
If the `start` and `pause` function only has this line different, I think it's better to define a common function and then apply different params to it.